scripts/benchmark_ad.py

```diff
@@ -0,0 +1,130 @@
+
+    required_args = ['targets']  # List any required arguments here
+
+    missing_args = [arg for arg in required_args if getattr(args, arg) is None]
+    if missing_args:
+        parser.error(
+            f'The following arguments are required: {", ".join(missing_args)}')
+
+    return args
+
+
+def run_benchmark(model_path: str, target_categories: List[str], should_overwrite: bool = False):
+    # Load the model
+    loader = ModelLoader()
+    model_name = loader.get_model_name('256x256_uncond', path=model_path)
+    logger.log(f"Loading model {model_name}...")
+    model, diffusion = loader.get_model('256x256_uncond', path=model_path)
+
+    # Create the components
+    logger.log("Creating benchmark components and trainer...")
+    noiser = TimestepUniformNoiser(diffusion)
+    denoiser = ModelTimestepUniformDenoiser(model, diffusion)
+    anomaly_map_generator = BatchFilteredSquaredError()
+    anomaly_scorer = MaxValueAnomalyScorer()
+
+    diffusion_ad = DiffusionAD(
+        noiser,
+        denoiser,
+        anomaly_map_generator,
+        anomaly_scorer,
+        DIFFUSION_AD_HPARAMS,
+        model_name=model_name)
+
+    # Create a PyTorch Lightning trainer for each target
+    trainer = {
+        target: pl.Trainer.from_argparse_args(diffusion_ad.args,
+                                              default_root_dir=os.path.join(
+                                                  diffusion_ad.args.root_output_dir, target),
+                                              max_epochs=diffusion_ad.args.num_epochs,
+                                              accelerator='gpu',
+                                              devices=1)
+        for target in target_categories
+    }
+
+    remaining_categories = diffusion_ad.get_remaining_categories()
+    logger.log(f'Target categories: {target_categories}\n'
+               f'Remaining categories: {target_categories & remaining_categories}\n'
+               f'Overwrite: {should_overwrite}')
+
+    for category in target_categories:
+        if should_overwrite or category in remaining_categories:
+            logger.log('\033[4m\033[1m' +
+                       f'Evaluating category: {category}' + '\033[0m')
+            diffusion_ad.args.category = category
+            try:
+                trainer[category].test(diffusion_ad)
+
+            except FileNotFoundError as e:
+                logger.log(
+                    e, f'\nFiles missing for category {category}! Please fix your dataset folder.\n')
+
+
+def main():
+    # Setup logger
+    logger.configure()
+
+    # Parse command-line arguments
+    args = parse_arguments()
+
+    # Load the model based on the provided path or default value
+    model_path: str = args.model
+
+    # Get the target categories
+    target_categories: List[str] = args.targets
+
+    # Update the DIFFUSION_AD_HPARAMS dictionary
+    DIFFUSION_AD_HPARAMS.reconstruction_batch_size = args.reconstruction_batch_size
```

Comment:
huh? no better way to inject this value?

Reply:
Thats a great way to inject it, as this is exactly this dict's purpose in life

Reply:
no it's a bad way but not enough for me to reject it